scripts/build.sh

```diff
@@ -182,8 +182,11 @@ if [ "$PLATFORM" = Darwin ] ; then
 	if [ -z "$BASESDK" ] ; then
 		BASESDK=$(ls $SDKROOT | grep -E "MacOSX11\.[0-9]+\.sdk" | head -1 | sed -e "s/MacOSX//;s/\.sdk//")
 		if [ -z "$BASESDK" ] ; then
-			echo "Cannot find a base SDK of type 10.x or 11.x under the SDK root of ${SDKROOT}"
-			exit 1;
+			BASESDK=$(ls $SDKROOT | grep -E "MacOSX12\.[0-9]+\.sdk" | head -1 | sed -e "s/MacOSX//;s/\.sdk//")
```

Comment:
so... we aren't looking for a 10.x one anymore, so the error message is wrong.
And wouldn't it be easier to just do
```
      BASESDK=$(ls $SDKROOT | grep -E "MacOSX1[1-6]\.[0-9]+\.sdk" | head -1 | sed -e "s/MacOSX//;s/\.sdk//")
```
(ok, that's looking forward to be ready for macOS versions 12-16... 🤣 -- use whatever end value feels reasonable)

Reply:
LOL. My regex foo is old and rusty. That looks a lot cleaner. I'll change it and try again.

I think we were still looking for SDK version 10.1X, though, because line 181 set the initial BASESDK search to:

       BASESDK=$(ls $SDKROOT | grep "MacOSX10\.1.\.sdk" | head -1 | sed -e "s/MacOSX//;s/\.sdk//")

Reply:
> I think we were still looking for SDK version 10.1X, though, because line 181 set the initial BASESDK search to:
> 
> ```
>    BASESDK=$(ls $SDKROOT | grep "MacOSX10\.1.\.sdk" | head -1 | sed -e "s/MacOSX//;s/\.sdk//")
> ```

Ahh - I didn't look for what we did earlier in the code... and I now remember why it is done that way... because I wanted to use the oldest one that was there... which is NOT what Apple wants you to do, anyway.
I think having this all in a single regexp makes more sense.

Reply:
If it is still the case that we want to use the oldest available SDK (including 10.X), should we look at using:

    BASESDK=$(ls $SDKROOT | grep -E "MacOSX1[0-6]\.[0-9]+\.sdk" | head -1 | sed -e "s/MacOSX//;s/\.sdk//")

That way we will still capture any remaining 10.X versions of the SDK.

Reply:
OK, now I am really confused. The regex "MacOSX1[0-6]\.[0-9]+\.sdk" does not seem to be matching things like MacOSX12.1.sdk or MacOSX13.0.sdk, but it the regex "MacOSX1[0-6]\.[0-9]*\.sdk" does match. I thought the + meant one or more occurrences of the previous expression.

`junkins@Dougs-MacBook-Pro scripts % ls -l /Library/Developer/CommandLineTools/SDKs                                
total 0
lrwxr-xr-x  1 root  wheel   14 Nov 19 18:16 MacOSX.sdk -> MacOSX13.0.sdk
drwxr-xr-x  7 root  wheel  224 Nov 19  2021 MacOSX12.1.sdk
drwxr-xr-x  7 root  wheel  224 Nov 19 18:16 MacOSX12.3.sdk
lrwxr-xr-x  1 root  wheel   14 Nov 19 18:15 MacOSX12.sdk -> MacOSX12.3.sdk
drwxr-xr-x  7 root  wheel  224 Oct  6 10:55 MacOSX13.0.sdk
lrwxr-xr-x  1 root  wheel   14 Nov 19 18:15 MacOSX13.sdk -> MacOSX13.0.sdk
junkins@Dougs-MacBook-Pro scripts % ls /Library/Developer/CommandLineTools/SDKs | grep "MacOSX1[0-6]\.[0-9]+\.sdk"
junkins@Dougs-MacBook-Pro scripts % ls /Library/Developer/CommandLineTools/SDKs | grep "MacOSX1[0-6]\.[0-9]*\.sdk"
MacOSX12.1.sdk
MacOSX12.3.sdk
MacOSX13.0.sdk
junkins@Dougs-MacBook-Pro scripts % 
`

Reply:
Nevermind. It was the lack of the -E option on grep that was causing it to not work with extended regexes.